### PR TITLE
feat: lint against remote baseBranch if user is on baseBranch

### LINKED ISSUE #13: Run linters on the base branch
Run linters on the base branch if the commit is pushed from the base branch. 

`The idea is to find the diff files between base_branch and origin/base_branch and run the linters for those files`

### PR DESCRIPTION
**Pull request by ericrallen**
I wasn't sure how you'd want to approach this, but here's a PR that implements opting into linting of the base branch so that no one already using this has to adjust their workflow at all.

Let me know if you have a different vision for the approach and I'd be happy to adjust the PR to reflect that.

Closes #13

### CODE REVIEW

**Comment by theenadayalank** [comment]
I think you have misunderstood the issue. The idea is not to opt-in linting the base branch. We should always run tasks regardless of the branch. 

The current process is If the changes have been pushed from the `base branch` (say, `master`) `lint-prepush` tries to find out the file diff between `current branch` and `base branch` (`master` against `master`) which are same in our case. So, this will not return any file list. 

We can let the `lint-prepush` to find the file diff between `the base branch` and `the origin/baseBranch` which in turn returns the diff file list.

**Comment by ericrallen** [comment]
@theenadayalank This was a proposal for not changing the exposed API and needing a major version upgrade due to the change potentially being something that could break someone's already established workflow.

I can easily remove the opt-in if you'd prefer.

In my brief tests locally this seemed to work as expected when opting-in, but I'll revisit it and add the relevant git remote to the diff if the base branch and current branch are the same.

**Comment by ericrallen** [comment]
@theenadayalank I've updated the PR to use the remote `baseBranch` if the user is on the `baseBranch` and have removed the opt-in functionality and language.

File: index.js [hunk 1]
```diff
@@ -56,30 +79,32 @@ if (process.stdout.isTTY) {
...
-                  concurrent: true,
-                  collapse: false
-                })
-                  .run()
-                  .then(() => {
-                    cache.setSync("linted-hash", commitHash);
-                    debug('Cached Current Commit Hash');
-                    log(success("\nVoila! 🎉  Code is ready to be Shipped.\n"));
+              resolveBaseBranch.then((base) => {
+                // Fetching committed git files
```
Comment (theenadayalank) [inline index.js]: logging the value of `base` would ease the debugging process

Comment (ericrallen) [reply index.js]: I haven't forgotten about this. Just got busy and haven't had a chance to circle back to it and make the update yet.

File: utils/fetchGitDiff.js [hunk 2]
```diff
@@ -2,8 +2,9 @@ const { execChildProcess } = require("./common");
 
 module.exports = function fetchGitDiff( baseBranch = "master" ) {
 
-  // git command to pull out the changed file names between current branch and base branch (Excluded delelted files which cannot be fetched now)
-  let command = `git diff --relative --name-only --diff-filter=d ${baseBranch}...HEAD`;
+  // git command to pull out the changed file names between current branch and base branch
+  // (Excluded deleted files which cannot be fetched now)
+  let command = `git diff --relative --name-only --diff-filter=d ${baseBranch}`;
```
Comment (theenadayalank) [inline utils/fetchGitDiff.js]: https://github.com/theenadayalank/lint-prepush/pull/30#pullrequestreview-256195647

**Comment by theenadayalank** [comment]
Closing this PR in favor of https://github.com/theenadayalank/lint-prepush/pull/67

**Pull request closed without merging**